device_backends/src/InterruptControllerHandler.cc

```diff
@@ -0,0 +1,130 @@
+// SPDX-FileCopyrightText: Deutsches Elektronen-Synchrotron DESY, MSK, ChimeraTK Project <chimeratk-support@desy.de>
+// SPDX-License-Identifier: LGPL-3.0-or-later
+#include "InterruptControllerHandler.h"
+
+#include "Axi4_Intc.h"
+#include "DummyIntc.h"
+#include "TriggerDistributor.h"
+#include "TriggeredPollDistributor.h"
+
+#include <tuple>
+
+namespace ChimeraTK {
+  //*********************************************************************************************************************/
+  InterruptControllerHandlerFactory::InterruptControllerHandlerFactory(DeviceBackend* backend) : _backend(backend) {
+    // we already know about the build-in handlers
+    _creatorFunctions["AXI4_INTC"] = Axi4_Intc::create;
+    _creatorFunctions["dummy"] = DummyIntc::create;
+  }
+
+  //*********************************************************************************************************************/
+  void InterruptControllerHandlerFactory::addControllerDescription(
+      std::vector<uint32_t> const& controllerID, std::string const& name, std::string const& description) {
+    _controllerDescriptions[controllerID] = {name, description};
+  }
+
+  //*********************************************************************************************************************/
+  boost::shared_ptr<InterruptControllerHandler> InterruptControllerHandlerFactory::createInterruptControllerHandler(
+      std::vector<uint32_t> const& controllerID, boost::shared_ptr<TriggerDistributor> parent) {
+    assert(!controllerID.empty());
+    std::string name, description;
+    try {
```

Comment:
I prefer try-blocks to be as small as possible in such cases, since it will catch the specific error only. Otherwise a `std::out_of_range` thrown at a different place might have a weired effect instead resulting in a prominent error.

Reply:
I agree with @mhier. I will keep the scope as limited as possible.